pkg/volume/plugins.go

```diff
@@ -131,26 +133,51 @@ type VolumePluginMgr struct {
 
 // Spec is an internal representation of a volume.  All API volume types translate to Spec.
 type Spec struct {
-	Name                   string
-	VolumeSource           api.VolumeSource
-	PersistentVolumeSource api.PersistentVolumeSource
-	ReadOnly               bool
+	Name             string
+	Volume           *api.Volume
```

Comment:
I don't understand this.  The only thing api.Volume has that api.VolumeSource doesn't is a name, which you have 1 line above...